BrainPortal/lib/boutiques_dir_maker.rb

```diff
@@ -0,0 +1,166 @@
+
+#
+# CBRAIN Project
+#
+# Copyright (C) 2008-2022
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# Some tools take expect that a file/directory input is
+# placed input a parent directory.
+#
+# For example we have the following Boutiques command-line:
+#
+#     apptool [PRECOMPUTED_INPUT]
+#
+# with the following input in Boutiques:
+#
+#     {
+#       "description": "A directory that contain the precomputed data process by...",
+#       "id": "precomputed_input",
+#       "name": "Precomputed input",
+#       "optional": false,
+#       "type": "File",
+#       "value-key": "[PRECOMPUTED_INPUT]"
+#     }
+#
+# and in the `cbrain:integrator_modules` section:
+#
+#     "BoutiquesDirMaker": {
+#         "precomputed_input": "fake_parent_dir"
+#      },
+#
+# In CBRAIN the user will select a userfile for example `sub-n`,
+# the final command line will become:
+#
+#     apptool precomputed/sub-n
+#
+module BoutiquesDirMaker
+
+  # Note: to access the revision info of the module,
+  # you need to access the constant directly, the
+  # object method revision_info() won't work.
+  Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
+
+  ############################################
+  # Portal Side Modifications
+  ############################################
+
+  # Adjust the description for the input with the mention of
+  # the fake parent directory
+  def descriptor_for_form #:nodoc:
+    descriptor = super.dup
+    boutiques_dir_maker_list = descriptor.custom_module_info('BoutiquesDirMaker')
+
+    boutiques_dir_maker_list.each do |inputid,dirname|
+      # Adjust the description
+      input        = descriptor.input_by_id(inputid)
+      description  = input.description.presence || ""
+      description += "\n" if description
```

Comment:
to avoid endofline in the beginning of  description, either ```description += "\n" if description.present?```

Reply:
or use following hack to avoid conditionals at all

```
    
      # Adjust the description
      input        = descriptor.input_by_id(inputid)
      new_paragraphs =   "This input will be copied in a parent folder `#{dirname}`.\n The parent folder will be used in the command line"
      input.description  = [input.description.presence,  new_paragraphs].compact.join('\n')

Reply:
or, with save navigation operator
```
      # Adjust the description
      input        = descriptor.input_by_id(inputid)
      new_paragraphs =   "This input will be copied in a parent folder `#{dirname}`.\n The parent folder will be used in the command line"
      input.description  = input.description.presence&.append(\'n')  + new_paragraphs
```